leap_ec/ops.py

```diff
@@ -366,30 +364,42 @@ def uniform_crossover(next_individual: Iterator,
         *Proceedings of the 4th international conference on genetic algorithms.* Morgan Kaufmann Publishers, 1991.
 
     >>> from leap_ec.individual import Individual
-    >>> from leap_ec.ops import uniform_crossover
+    >>> from leap_ec.ops import uniform_crossover, naive_cyclic_selection
     >>> import numpy as np
 
     >>> genome1 = np.array([0, 0])
     >>> genome2 = np.array([1, 1])
     >>> first = Individual(genome1)
     >>> second = Individual(genome2)
-    >>> i = iter([first, second])
-    >>> result = uniform_crossover(i)
-
+    >>> pop = [first, second]
+    >>> select = naive_cyclic_selection(pop)
+    >>> op = uniform_crossover()
+    >>> result = op(select)
     >>> new_first = next(result)
     >>> new_second = next(result)
 
     The probability can be tuned via the `p_swap` parameter:
+    >>> op = uniform_crossover(p_swap=0.1)
+    >>> result = op(select)
+    
+    If `persist_children` is True and there is a child that was made by crossover but isn't
+    used in the first call, it will be yielded in a future call.
+    
+    >>> op = uniform_crossover(p_xover=0.0, persist_children=True)
+    >>> next(op(select)) is first
+    True
+    >>> next(op(select)) is second
+    True
```

Comment:
I think this test will pass even if `persist_children=False`?  So we'll need to modify it to specifically target what is dfferent about the `persist_children` behavior.

Reply:
Nope, just tested it! I'll try to make it clearer, but the important part is that `op(select)` is being called twice. This causes the issue where the second yielded individual is never consumed, since it restarts the loop over again.